code/modules/reagents/chemistry/reagents/alcohol_reagents.dm

```diff
@@ -1797,12 +1797,13 @@ All effects don't start immediately, but rather get worse over time; the rate is
 	glass_name = "glass of [name]"
 	glass_desc = description
 	for(var/taste in tastes)
-		if(tastes[taste] < minimum_percent)
-			continue
-		if(tastes[taste] >= minimum_percent*2)
-			primary_tastes += taste
-		else if(tastes[taste] >= minimum_percent)
-			secondary_tastes += taste
+		switch(tastes[taste])
+			if(-INFINITY to minimum_percent)
+				continue
```

Comment:
You don't need this check for under minimum_percent if you have the others. Also if it's exactly equal to minimum percent it will skip unlike how you had it before.